Tests/WitsmlExplorer.Api.Tests/Workers/ReplaceObjectsWorkerTests.cs

```diff
@@ -0,0 +1,201 @@
+
+using Microsoft.Extensions.Logging;
+
+using Moq;
+
+using Witsml;
+using Witsml.Data;
+using Witsml.Data.Tubular;
+using Witsml.ServiceReference;
+
+using WitsmlExplorer.Api.Jobs;
+using WitsmlExplorer.Api.Jobs.Common;
+using WitsmlExplorer.Api.Models;
+using WitsmlExplorer.Api.Services;
+using WitsmlExplorer.Api.Workers;
+using WitsmlExplorer.Api.Workers.Copy;
+using WitsmlExplorer.Api.Workers.Delete;
+
+using Xunit;
+
+namespace WitsmlExplorer.Api.Tests.Workers
+{
+    public class ReplaceObjectsWorkerTests
+    {
+        private readonly Mock<IWitsmlClientProvider> _witsmlClientProvider;
+        private readonly ReplaceObjectsWorker _replaceObjectWorker;
+        private readonly CopyObjectsWorker _copyObjectsWorker;
+        private readonly DeleteObjectsWorker _deleteObjectsWorker;
+        private readonly Mock<IWitsmlClient> _witsmlClient;
+
+        private const string WellUid = "wellUid";
+        private const string SourceWellboreUid = "sourceWellboreUid";
+        private const string TargetWellboreUid = "targetWellboreUid";
+        private const string ObjectUid = "objectUid";
+        private static readonly string[] ObjectUids = { "objectUid1", "objectUid2" };
+
+        public ReplaceObjectsWorkerTests()
+        {
+            _witsmlClientProvider = new();
+            _witsmlClient = new Mock<IWitsmlClient>();
+            _witsmlClientProvider.Setup(provider => provider.GetClient()).Returns(_witsmlClient.Object);
+            _witsmlClientProvider.Setup(provider => provider.GetSourceClient()).Returns(_witsmlClient.Object);
+
+            Mock<ILogger<CopyObjectsJob>> copyObjectsLogger = new();
+            CopyUtils copyUtils = new(new Mock<ILogger<CopyUtils>>().Object);
+            CopyLogWorker copyLogWorker = new(new Mock<ILogger<CopyObjectsJob>>().Object, _witsmlClientProvider.Object);
+            _copyObjectsWorker = new CopyObjectsWorker(copyObjectsLogger.Object, _witsmlClientProvider.Object, copyUtils, copyLogWorker);
+
+            ILogger<DeleteObjectsJob> deleteObjectsLogger = new Mock<ILogger<DeleteObjectsJob>>().Object;
+            _deleteObjectsWorker = new DeleteObjectsWorker(deleteObjectsLogger, _witsmlClientProvider.Object);
+            ILogger<ReplaceObjectsJob> replaceObjectsLogger = new Mock<ILogger<ReplaceObjectsJob>>().Object;
+            _replaceObjectWorker = new ReplaceObjectsWorker(replaceObjectsLogger, _copyObjectsWorker, _deleteObjectsWorker);
+        }
+
+        [Fact]
+        public async Task Execute_Delete_Part_ThrowsError()
+        {
+            SetUpStoreForCopy(true);
+            SetUpStoreForDelete(false);
+            ReplaceObjectsJob replaceObjectsJob = SetUpReplaceObjectsJob();
+            (WorkerResult workerResult, RefreshAction refreshAction) = await _replaceObjectWorker.Execute(replaceObjectsJob);
+            Assert.False(workerResult.IsSuccess);
+            Assert.Equal(" Failed to delete some WitsmlTubulars", workerResult.Message);
```

Comment:
If you find small bugs like this, It's better to add a quick fix for it than to expect something to be "wrong". I wouldn't expect the sentence to start with a space.

Reply:
fixed